python/ray/dashboard/client/src/pages/dashboard/logical-view/Actor.tsx

```diff
@@ -109,204 +97,195 @@ class Actor extends React.Component<Props & WithStyles<typeof styles>, State> {
+            "This does not imply that the objects are in active use or colocated on the node with the actor " +
+            `currently. This can be useful for debugging memory leaks. See the docs at ${memoryDebuggingDocLink} ` +
+            "for more information.",
+        },
+        {
+          label: "Number of local objects",
+          value: actor.numLocalObjects.toLocaleString(),
+          tooltip:
+            "The number of small objects that this actor has stored in its local in-process memory store. This can be useful for " +
+            `debugging memory leaks. See the docs at ${memoryDebuggingDocLink} for more information`,
+        },
+        {
+          label: "Object store memory used (MiB)",
+          value: actor.usedObjectStoreMemory.toLocaleString(),
+          tooltip:
+            "The total amount of memory that this actor is occupying in the Ray object store. " +
+            "If this number is increasing without bounds, you might have a memory leak. See " +
+            `the docs at: ${memoryDebuggingDocLink} for more information.`,
+        },
+      ]
+    : [
+        {
+          label: "Actor ID",
+          value: actor.actorId,
+          tooltip: "",
+        },
+        {
+          label: "Required resources",
+          value:
+            actor.requiredResources &&
+            Object.entries(actor.requiredResources).length > 0 &&
+            Object.entries(actor.requiredResources)
+              .sort((a, b) => a[0].localeCompare(b[0]))
+              .map(([key, value]) => `${value.toLocaleString()} ${key}`)
+              .join(", "),
+          tooltip: "",
+        },
+      ];
 
-    // Construct the custom message from the actor.
-    let actorCustomDisplay: JSX.Element[] = [];
-    if (isFullActorInfo(actor) && actor.webuiDisplay) {
-      actorCustomDisplay = Object.keys(actor.webuiDisplay)
-        .sort()
-        .map((key, _, __) => {
-          // Construct the value from actor.
-          // Please refer to worker.py::show_in_dashboard for schema.
-          const valueEncoded = actor.webuiDisplay![key];
-          const valueParsed = JSON.parse(valueEncoded);
-          let valueRendered = valueParsed["message"];
-          if (valueParsed["dtype"] === "html") {
-            valueRendered = (
-              <div
-                className={classes.inlineHTML}
-                dangerouslySetInnerHTML={{ __html: valueRendered }}
-              ></div>
-            );
-          }
+  // Construct the custom message from the actor.
+  let actorCustomDisplay: JSX.Element[] = [];
+  if (isFullActorInfo(actor) && actor.webuiDisplay) {
+    actorCustomDisplay = Object.keys(actor.webuiDisplay)
+      .sort()
+      .map((key, _, __) => {
+        // Construct the value from actor.
+        // Please refer to worker.py::show_in_webui for schema.
+        const valueEncoded = actor.webuiDisplay![key];
```

Comment:
If `key` is empty isn't `valueParsed` broken? I am not familiar with this `!` syntax, and it probably avoids that issue?

Reply:
It isn't broken because it won't be called. Key is actually coming from a map statement that iterates through the keys of actor.webuidisplay

Reply:
Ah, I see. That makes sense. Sorry for missing it!